compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -675,8 +675,8 @@ impl<'a> Parser<'a> {
         let precise_capturing = if self.eat_keyword(kw::Use) {
             let use_span = self.prev_token.span;
             self.psess.gated_spans.gate(sym::precise_capturing, use_span);
-            let args = self.parse_precise_capturing_args()?;
-            Some(P((args, use_span)))
+            let (args, args_span) = self.parse_precise_capturing_args()?;
+            Some(P((args, use_span.to(args_span))))
```

Comment:
I think something like

```rust
#![feature(precise_capturing)]

macro_rules! foo {
    ($a:tt $b:tt) => {
         fn hello(_: impl use $b $a Sized) {}
    };
}

foo!(><);
```

will create very funny diagnostics now

Reply:
I feel like this can happen with a lot of syntax -- do you have an idea of how to fix this? Because otherwise I kinda would rather just leave it 😆

Reply:
yeaaa... no idea :D